### PR TITLE
Add examples

### PR DESCRIPTION
**Pull request by tshedor**
Examples for brick_core and the repo

### CODE REVIEW

File: example/lib/app/models/pizza.dart [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  ),
+)
+class Pizza extends OfflineFirstWithRestModel {
+  @Sqlite(unique: true)
+  final int id;
+
+  @Rest(enumAsString: true)
+  final List<Topping> toppings;
+
+  @OfflineFirst(where: {'id': "data['customer_id']"})
```
Comment (jnhuynh) [inline example/lib/app/models/pizza.dart]: Given this example, would be it make sense to add some comments, what each annotation does? E.G. sqlite is sqlite serdes only, rest is for rest serdes config, and offline first is whatever it is (I have no idea actually).

File: packages/brick_core/example/example.dart [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    ModelRepository<FileModel> repository,
+  });
+  Future<String> toFile(
+    _Model instance, {
+    FileProvider provider,
+    ModelRepository<FileModel> repository,
+  });
+}
+
+/// This is generated
```
Comment (jnhuynh) [inline packages/brick_core/example/example.dart]: Does it make sense to put the generated into its own file so we reduce confusion?

File: packages/brick_core/example/example.dart [hunk 3]
```diff
@@ -0,0 +1,121 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:glob/glob.dart';
+
+import '../lib/core.dart';
+
+abstract class FileModel extends Model {
```
Comment (jnhuynh) [inline packages/brick_core/example/example.dart]: Great choice for a provider example!

File: packages/brick_core/example/example.dart (hunk 2 above)
Comment (jnhuynh) [reply packages/brick_core/example/example.dart]: Also does this generation actually happen without an extension to brick build?

File: packages/brick_core/example/example.dart (hunk 3 above)
Comment (ericwindmill) [reply packages/brick_core/example/example.dart]: +1... it is very helpful

File: packages/brick_core/example/example.dart (hunk 2 above)
Comment (tshedor) [reply packages/brick_core/example/example.dart]: No, I added a comment. I still think it's important to include the adapter to show, at the lowest-level, what Brick is doing and what it expects. If you're in this example, you're likely creating a provider.

**Pull request merged**